File: win-prog/ProgWinVB/Buttons and Labels and Scrolls (Oh, My!)/TwoButtonsAnchor/TwoButtonsAnchor.vb
'-------------------------------------------------
' TwoButtonsAnchor.vb (c) 2002 by Charles Petzold
'-------------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class TwoButtonsAnchor
    Inherits Form

    Shared Sub Main()
        Application.Run(New TwoButtonsAnchor())
    End Sub

    Sub New()
        Text = "Two Buttons with Anchor"
        ResizeRedraw = True

        Dim cxBtn As Integer = 5 * Font.Height
        Dim cyBtn As Integer = 2 * Font.Height
        Dim dxBtn As Integer = Font.Height

        Dim btn As New Button()
        btn.Parent = Me
        btn.Text = "&Larger"
        btn.Location = New Point(dxBtn, dxBtn)
        btn.Size = New Size(cxBtn, cyBtn)
        AddHandler btn.Click, AddressOf ButtonLargerOnClick

        btn = New Button()
        btn.Parent = Me
        btn.Text = "&Smaller"
        btn.Location = New Point(ClientSize.Width - cxBtn - dxBtn, _
                                 ClientSize.Height - cyBtn - dxBtn)
        btn.Size = New Size(cxBtn, cyBtn)
        btn.Anchor = AnchorStyles.Right Or AnchorStyles.Bottom
        AddHandler btn.Click, AddressOf ButtonSmallerOnClick
    End Sub

    Private Sub ButtonLargerOnClick(ByVal obj As Object, ByVal ea As EventArgs)
        Left -= CInt(0.05 * Width)
        Top -= CInt(0.05 * Height)
        Width += CInt(0.1 * Width)
        Height += CInt(0.1 * Height)
    End Sub

    Private Sub ButtonSmallerOnClick(ByVal obj As Object, ByVal ea As EventArgs)
        Left += CInt(Width / 22)
        Top += CInt(Height / 22)
        Width -= CInt(Width / 11)
        Height -= CInt(Height / 11)
    End Sub
End Class

File: win-prog/ProgWinCS/Buttons and Labels and Scrolls (Oh, My!)/TwoButtonsAnchor/TwoButtonsAnchor.cs
//-----------------------------------------------
// TwoButtonsAnchor.cs © 2001 by Charles Petzold
//-----------------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class TwoButtonsAnchor: Form
{
     public static void Main()
     {
          Application.Run(new TwoButtonsAnchor());
     }
     public TwoButtonsAnchor()
     {
          Text = "Two Buttons with Anchor";
          ResizeRedraw = true;

          int cxBtn = 5 * Font.Height;
          int cyBtn = 2 * Font.Height;
          int dxBtn =     Font.Height;

          Button btn = new Button();
          btn.Parent   = this;
          btn.Text     = "&Larger";
          btn.Location = new Point(dxBtn, dxBtn);
          btn.Size     = new Size(cxBtn, cyBtn);
          btn.Click   += new EventHandler(ButtonLargerOnClick);
          
          btn = new Button();
          btn.Parent   = this;
          btn.Text     = "&Smaller";
          btn.Location = new Point(ClientSize.Width  - cxBtn - dxBtn,
                                   ClientSize.Height - cyBtn - dxBtn);
          btn.Size     = new Size(cxBtn, cyBtn);
          btn.Anchor   = AnchorStyles.Right | AnchorStyles.Bottom;
          btn.Click   += new EventHandler(ButtonSmallerOnClick);
     }
     void ButtonLargerOnClick(object obj, EventArgs ea)
     {
          Left   -= (int)(0.05 * Width);
          Top    -= (int)(0.05 * Height);
          Width  += (int)(0.10 * Width);
          Height += (int)(0.10 * Height);
     }
     void ButtonSmallerOnClick(object obj, EventArgs ea)
     {
          Left   += (int)(Width  / 22f);
          Top    += (int)(Height / 22f);
          Width  -= (int)(Width  / 11f);
          Height -= (int)(Height / 11f);
     }
}
